usdigitalresponse/grants-ingest
# Enqueue FFIS email downloads

### Ticket #4 
_Closes #4_

## Description
As described in the ticket,
- Defines a new Lambda, EnqueueFFISDownload
- Defines SQS queue

Handler (in `cmd/EnqueueFFISDownload`) responds to S3 events in the grants bucket source, looks for a download link in it, and enqueues it into SQS for later processing.

Some things to discuss:
1. ~I'm not familiar with the CloudPosse modules, but I didn't see where SES could be configured for the SQS queue, so I used the stock TF resource~
2. ~For the TF structure, I put the queue in the EnqueueFFISDownload module. Should it be top-level?~
3. ~Am I using the log formatter  correctly?~

I expect some revs on this, but it was getting lengthy and I wanted to come up for air/iterate. Appreciate feedback as always!

## Testing 
1. Launch localstack (`docker compose up -d`) works here
5. Apply the infrastructure `tflocal apply -var-file=local.tfvars  -auto-approve`
6. Push a test file to the bucket `awslocal s3 cp ../cmd/EnqueueFFISDownload/fixtures/good.eml s3://grants-ingest-grantssourcedata-000000000000-us-west-2/sources/2023/04/24/ffis/raw.eml`
7. Observe the message in SQS via one of the [supported methods](https://docs.localstack.cloud/user-guide/aws/sqs/), for example `curl -H "Accept: application/json" http://localhost:4566/_aws/sqs/messages?QueueUrl=http://localhost:4566/000000000000/ffis_downloads | jq '.ReceiveMessageResponse.ReceiveMessageResult.Message.Body'`

### Automated and Unit Tests

### Manual tests for Reviewer

## Checklist
